module/VuFindConsole/src/VuFindConsole/Command/Menu/MenuCommand.php

```diff
@@ -0,0 +1,413 @@
+<?php
+
+/**
+ * Console command: interactive menu.
+ *
+ * PHP version 8
+ *
+ * Copyright (C) Villanova University 2024.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Console
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+
+namespace VuFindConsole\Command\Menu;
+
+use Exception;
+use Symfony\Component\Console\Attribute\AsCommand;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Question\ChoiceQuestion;
+use Symfony\Component\Console\Question\Question;
+use VuFindConsole\Command\PluginManager;
+
+use function count;
+use function in_array;
+
+/**
+ * Console command: interactive menu.
+ *
+ * @category VuFind
+ * @package  Console
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+#[AsCommand(
+    name: 'menu/menu',
+    description: 'Interactive menu'
+)]
+class MenuCommand extends Command
+{
+    /**
+     * Menu option for running a command.
+     *
+     * @var string
+     */
+    protected string $runCommand = 'Run Command';
+
+    /**
+     * Menu option for exiting a command without running it.
+     *
+     * @var string
+     */
+    protected string $exitCommand = 'Exit Command';
+
+    /**
+     * Constructor
+     *
+     * @param array         $config         Menu configuration
+     * @param PluginManager $commandManager Plugin manager for internal console commands
+     * @param string|null   $name           The name of the command; passing null means it must
+     * be set in configure()
+     */
+    public function __construct(protected array $config, protected PluginManager $commandManager, $name = null)
```

Comment:
```suggestion
    public function __construct(protected array $config, protected PluginManager $commandManager, ?string $name = null)
```
`$name` can also have a type declaration.